examples/multimodal/components/worker.py

```diff
@@ -0,0 +1,161 @@
+)
+from vllm.inputs.data import TokensPrompt
+from vllm.sampling_params import RequestOutputKind
+
+from dynamo.llm import KvMetricsPublisher
+from dynamo.sdk import async_on_start, depends, dynamo_context, dynamo_endpoint, service
+
+logger = logging.getLogger(__name__)
+
+
+@service(
+    dynamo={
+        "enabled": True,
+        "namespace": "dynamo",
+    },
+    resources={"gpu": 1, "cpu": "10", "memory": "20Gi"},
+    workers=1,
+)
+class VllmWorker:
+    encode_worker = depends(EncodeWorker)
+
+    def __init__(self):
+        self.client = None
+        self.min_workers = 1
+        class_name = self.__class__.__name__
+        self.engine_args = parse_vllm_args(class_name, "")
+        self.do_remote_prefill = self.engine_args.remote_prefill
+        self.model_name = (
+            self.engine_args.served_model_name
+            if self.engine_args.served_model_name is not None
+            else "vllm"
+        )
+
+        if self.engine_args.remote_prefill:
+            logger.warning(
+                "This example uses local prefill, setting remote prefill to False"
+            )
+            self.engine_args.remote_prefill = False
+
+        self.metrics_publisher = KvMetricsPublisher()
+
+        signal.signal(signal.SIGTERM, self.shutdown_vllm_engine)
+        signal.signal(signal.SIGINT, self.shutdown_vllm_engine)
+
+    @async_on_start
+    async def async_init(self):
+        self._engine_context = build_async_engine_client_from_engine_args(
+            self.engine_args
+        )
+        if self._engine_context is not None:
+            self.engine_client = await self._engine_context.__aenter__()
+        else:
+            raise RuntimeError("Failed to initialize engine client")
+
+        if self.engine_args.router == "kv":
+            logger.info(
+                "Multimodal requests are not supported for kv router mode, defaulting to round-robin"
+            )
+            self.engine_args.router = "round-robin"
+
+        runtime = dynamo_context["runtime"]
+
+        enc_comp_ns, enc_comp_name = EncodeWorker.dynamo_address()  # type: ignore
+        self.encode_worker_client = (
+            await runtime.namespace(enc_comp_ns)
+            .component(enc_comp_name)
+            .endpoint("encode")
+            .client()
+        )
+
+        await check_required_workers(self.encode_worker_client, self.min_workers)
+
+        self.disaggregated_router = None
+        logger.info("VllmWorker has been initialized")
+
+    def shutdown_vllm_engine(self, signum, frame):
+        """Shutdown the background loop"""
+        logger.info(f"Received signal {signum}, shutting down")
+        loop = asyncio.get_event_loop()
+        try:
+            self.engine_client.close()
+            logger.info("VllmWorker shutdown complete")
+        except Exception as e:
+            logger.error(f"Error during shutdown: {e}")
+        finally:
+            loop.stop()
+
+    async def create_metrics_publisher_endpoint(self):
```

Comment:
question: is this used?

Reply:
No used at the moment. Removed. Thanks for the catch.